book/optimization.qmd

```diff
@@ -422,47 +632,86 @@ Nested resampling is not restricted to hyperparameter tuning.
 You can swap the `r ref("AutoTuner")` for an `r ref("AutoFSelector")` and estimate the performance of a model which is fitted on an optimized feature subset.
 
 With the created `r ref("ResampleResult")` we can now inspect the executed resampling iterations more closely.
-See the section on [Resampling](#resampling) for more detailed information about `r ref("ResampleResult")` objects.
+See the section on resampling (@sec-resampling) for more detailed information about `r ref("ResampleResult")` objects.
 
 The `extract_inner_tuning_results()` function prints the three optimized hyperparameter configurations and the corresponding performance estimates on the inner resampling.
-We observe a trend toward small `cp` and `minsplit` values.
+We observe a trend toward large `cost` and small `gamma` values.
 Keep in mind that these values should not be used to fit a final model.
+The selected hyperparameters might differ greatly between the resampling iterations.
+On the one hand, this could be due to the optimization algorithm used.
+For simple algorithms like random search, we don't expect stable hyperparameters but more advanced methods like irace converge to an optimal hyperparameter configuration.
+On the other hand, small data sets and a low number of resampling iterations might introduce too much randomness for stable hyperparameters.
 
-```{r optimization-028}
-#| eval: false
-extract_inner_tuning_results(rr)
-```
-
-```{r optimization-029}
-#| echo: false
-extract_inner_tuning_results(rr)[, c("iteration", "cp",  "minsplit", "classif.ce")]
+```{r optimization-042}
+extract_inner_tuning_results(rr)[, list(iteration, cost, gamma, classif.ce)]
 ```
 
 Now we want to compare the predictive performances estimated on the outer resampling to the inner resampling.
 Significantly lower predictive performances on the outer resampling indicate that the models with the optimized hyperparameters overfit the data.
 
-```{r optimization-030}
-#| eval: false
-rr$score()
-```
-
-```{r optimization-031}
-#| echo: false
-rr$score()[, c("iteration", "classif.ce"), with = FALSE]
+```{r optimization-043}
+rr$score()[, list(iteration, classif.ce)]
 ```
 
 We see a slightly over-optimistic performance estimation on the inner resampling.
 
+If we want to examine the results even more in detail, we can also display the inner archives.
```

Comment:
Please can you check this? What do you mean by lower performance? Lower CE? Is this overfitting?